pkgs/os-specific/linux/igh-ethercat-master/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib, stdenv, fetchFromGitLab, autoconf, automake, pkg-config, libtool, kernel ? null }:
+
+stdenv.mkDerivation
+{
+  pname = "igh-ethercat-master";
+  version = "1.5-${kernel.modDirVersion}";
+
+  src = fetchFromGitLab {
+    owner = "etherlab.org";
+    repo = "ethercat";
+    rev = "4933150c17139e5e469197603984ca41ef9dbebe";
+    sha256 = "sha256-f60MWcbmOGe1Zflig9gxd52HVD8hyJ9AdopCUbsfecg=";
+  };
+
+  nativeBuildInputs = [ autoconf automake pkg-config libtool ] ++ kernel.moduleBuildDependencies;
+
+  makeFlags = [ "KDIR='${kernel.dev}/lib/modules/${kernel.modDirVersion}/build'" ];
+
+  patches = [
+    ./ethercatctl.in.patch
+  ];
+
+  buildPhase = ''
+    bash ./bootstrap
+    ./configure --prefix=$out --with-linux-dir="${kernel.dev}/lib/modules/${kernel.modDirVersion}/build/" --disable-8139too --disable-e100 --disable-e1000 --disable-e1000e --disable-r8169 --disable-igb --enable-ccat
+
+    # Copy configured build directory, to use with erhetcat-rs (https://github.com/ethercat-rs/ethercat)
+    mkdir -p $out/build/
+    cp -r * $out/build/
+
+    make all modules
```

Comment:
This should use buildFlags/makeFlags

Reply:
I put the `all modules` in makeFlags instead, and moved the mkdir/cp to configurePhase.